org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandler.java

```diff
@@ -394,6 +399,10 @@ private synchronized void doNodeInitialisation() {
                     stateDescriptions.put(channel.getUID(), stateDescription);
                 }
             }
+
+            if (initializeDevice) {
+                thing.setProperty(THING_PROPERTY_DEVICE_INITIALIZED, Boolean.TRUE.toString());
```

Comment:
It seems possible to incorrectly report that the device is initialised when it isn't?

This statement will always set the device_initialised to true - even if the device was not actually initialised. By this, I mean if there is an error above, you ignore the error and we still say the device is initialised.

Personally, I think that if the user, or system, wanted to initialise the device, and it fails, then we should not set this property to true. This would allow it to retry on next startup rather than leave the device unusable.

Reply:
You're right. I will change it and set the property to `true` only if all channels are initialized successfully.